metacat-metadata-mysql/src/main/java/com/netflix/metacat/metadata/mysql/MysqlUserMetadataInterceptorService.java

```diff
@@ -0,0 +1,167 @@
+/*
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package com.netflix.metacat.metadata.mysql;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.collect.Lists;
+import com.netflix.metacat.common.QualifiedName;
+import com.netflix.metacat.common.dto.DefinitionMetadataDto;
+import com.netflix.metacat.common.json.MetacatJson;
+import com.netflix.metacat.common.server.connectors.exception.InvalidMetadataException;
+import com.netflix.metacat.common.server.properties.Config;
+import com.netflix.metacat.common.server.usermetadata.MetadataInterceptor;
+import com.netflix.metacat.common.server.usermetadata.UserMetadataServiceException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.ResultSetExtractor;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * MysqlUserMetadataInterceptorService.
+ *
+ * @author zhenl
+ * @since 1.2.0
+ */
+@Slf4j
+@Transactional("metadataTxManager")
+public class MysqlUserMetadataInterceptorService extends MysqlUserMetadataService {
+    private final MetadataInterceptor metadataInterceptor;
+
+    /**
+     * Constructor.
+     *
+     * @param jdbcTemplate        jdbc template
+     * @param metacatJson         json utility
+     * @param metadataInterceptor business metadata manager
+     * @param config              config
+     */
+    public MysqlUserMetadataInterceptorService(
+        final JdbcTemplate jdbcTemplate,
+        final MetacatJson metacatJson,
+        final Config config,
+        final MetadataInterceptor metadataInterceptor
+    ) {
+        super(jdbcTemplate, metacatJson, config);
+        this.metadataInterceptor = metadataInterceptor;
+    }
+
+    @Nonnull
+    @Override
+    @Transactional(readOnly = true)
+    public Optional<ObjectNode> getDefinitionMetadata(
+        @Nonnull final QualifiedName name) {
+        final Optional<ObjectNode> retData = getJsonForKey(
```

Comment:
why are we not delegating the call to the method in the super class?

Reply:
Yes, we can. I will change it.